### ISSUE TITLE
Delegate_to & with_items & groups.all fails with "apt" module.

### ISSUE THREAD

**Issue by ghost** [ansible/ansible#13666]
# Overview

It seems delegate_to and with_items fails to properly split target hosts when using "apt" module.
The task is not divided between different members but delegated to an unexistent one (FQDN : host1,host2,hostN...) and so the hostname resolution fails.

I'm using Ansible 1.7.2
# Example tasks

This task fails everytime :

``` yaml
- name: Installing NRPE on hosts.
  apt:
      name=nagios-nrpe-server
  delegate_to: "{{ item }}"
  with_items: groups.all
```

This one works: 

``` yaml
- name: Installing NRPE on hosts.
  shell:
      apt-get install -y nagios-nrpe-server
  delegate_to: "{{ item }}"
  with_items: groups.all
```
# Playbook output

```
ansible-playbook -i hosts site.yml -vvvv --start-at-task "Installing NRPE on hosts."
[snip]
TASK: [shinken | Installing NRPE on hosts.] *********************************** 
<HOST1_FQDN,HOST2_FQDN> ESTABLISH CONNECTION FOR USER: root
<HOST1_FQDN,HOST2_FQDN> REMOTE_MODULE apt name=nagios-nrpe-server
<HOST1_FQDN,HOST2_FQDN> EXEC ['ssh', '-C', '-tt', '-vvv', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/home/REDACTED/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'User=root', '-o', 'ConnectTimeout=10', u'HOST1_FQDN,HOST2_FQDN', "/bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1450979972.33-158339587471846 && echo $HOME/.ansible/tmp/ansible-tmp-1450979972.33-158339587471846'"]
fatal: [mon.sk4.nz -> {{ item }}] => SSH encountered an unknown error. The output was:
OpenSSH_6.7p1 Debian-5, OpenSSL 1.0.1k 8 Jan 2015
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug1: auto-mux: Trying existing master
debug1: Control socket "/home/sk4nz/.ansible/cp/ansible-ssh-HOST1_FQDN,HOST2_FQDN-22-root" does not exist
ssh: Could not resolve hostname HOST1_FQDN,HOST2_FQDN: Name or service not known
```

**Comment by systeminsightsbuild** [comment]
Why is this closed? I am seeing the same problem in 1.9.4.

**Comment by Yannig** [comment]
Work with last devel version. Maybe you can give a try with Ansible v2.0.0.2.

```
ansible-playbook loop.yml -vvv
Using /etc/ansible/ansible.cfg as config file
1 plays in loop.yml

PLAY [Loop + apt] **************************************************************

TASK [apt] *********************************************************************
task path: /home/yannig/dev/yannig/ansible-issue-low-speed/apt+delegate+with/loop.yml:7
ESTABLISH LOCAL CONNECTION FOR USER: yannig
localhost EXEC /bin/sh -c '( umask 22 && mkdir -p "$( echo $HOME/.ansible/tmp/ansible-tmp-1454263933.54-126930707966436 )" && echo "$( echo $HOME/.ansible/tmp/ansible-tmp-1454263933.54-126930707966436 )" )'
localhost PUT /tmp/tmp_tkjvx TO /home/yannig/.ansible/tmp/ansible-tmp-1454263933.54-126930707966436/apt
localhost EXEC /bin/sh -c 'LANG=C LC_ALL=C LC_MESSAGES=C /usr/bin/python /home/yannig/.ansible/tmp/ansible-tmp-1454263933.54-126930707966436/apt; rm -rf "/home/yannig/.ansible/tmp/ansible-tmp-1454263933.54-126930707966436/" > /dev/null 2>&1'
localhost EXEC /bin/sh -c '( umask 22 && mkdir -p "$( echo $HOME/.ansible/tmp/ansible-tmp-1454263933.98-163831178836480 )" && echo "$( echo $HOME/.ansible/tmp/ansible-tmp-1454263933.98-163831178836480 )" )'
localhost PUT /tmp/tmpqL_u9_ TO /home/yannig/.ansible/tmp/ansible-tmp-1454263933.98-163831178836480/apt
localhost EXEC /bin/sh -c 'LANG=C LC_ALL=C LC_MESSAGES=C /usr/bin/python /home/yannig/.ansible/tmp/ansible-tmp-1454263933.98-163831178836480/apt; rm -rf "/home/yannig/.ansible/tmp/ansible-tmp-1454263933.98-163831178836480/" > /dev/null 2>&1'
ok: [localhost -> localhost] => (item=localhost) => {"cache_update_time": 0, "cache_updated": false, "changed": false, "invocation": {"module_args": {"_ansible_check_mode": false, "_ansible_debug": false, "_ansible_diff": false, "_ansible_no_log": false, "_ansible_verbosity": 3, "cache_valid_time": null, "deb": null, "default_release": null, "dpkg_options": "force-confdef,force-confold", "force": false, "install_recommends": null, "name": "bash", "package": ["bash"], "purge": false, "state": "present", "update_cache": false, "upgrade": null}, "module_name": "apt"}, "item": "localhost"}
ok: [localhost -> 127.0.0.2] => (item=127.0.0.2) => {"cache_update_time": 0, "cache_updated": false, "changed": false, "invocation": {"module_args": {"_ansible_check_mode": false, "_ansible_debug": false, "_ansible_diff": false, "_ansible_no_log": false, "_ansible_verbosity": 3, "cache_valid_time": null, "deb": null, "default_release": null, "dpkg_options": "force-confdef,force-confold", "force": false, "install_recommends": null, "name": "bash", "package": ["bash"], "purge": false, "state": "present", "update_cache": false, "upgrade": null}, "module_name": "apt"}, "item": "127.0.0.2"}

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0   
```
